demo/src/app/components/typeahead/overview/sections/basic-usage/template.html

```diff
@@ -0,0 +1,47 @@
+<p>
+  To set up a typeahead instance, you need to pass an initialization function to it. This function must return an <code>Observable</code> spawning lists of options: this is how you control the list of displayed options.
+</p>
+
+<p>
+  Pass it that way on the template side:
+</p>
+
+<ngbd-code [snippet]="snippets.observableSetup.template"></ngbd-code>
+
+<p>
+  And define it that way on the component side:
+</p>
+
+<ngbd-code [snippet]="snippets.observableSetup.component"></ngbd-code>
+
+
+
+<h5>The basic <code>Observable</code> chain</h5>
+
+<p>
+  In this basic example we will simply use a statically defined and synchronously available set of data — color names:
+</p>
+
+<ngbd-code [snippet]="snippets.data"></ngbd-code>
+
+<p>
+  The function used to provide the <code>Observable</code> to the typeahead receives a single argument: another <code>Observable</code>. It spawns:
+</p>
+<ul>
+  <li>what: strings corresponding to the current content of the input</li>
+  <li>when: on the <code>input</code> event of the input</li>
+</ul>
+
+<p>
+  You can therefore use this input as a search term and apply filtering for example:
+</p>
+
+<ngbd-code [snippet]="snippets.observable.mapping"></ngbd-code>
+
+<p>
+  Here we return all the colors when the search term is empty, otherwise we return all the colors starting with it.
```

Comment:
Describing a ternary operator with words...

Reply:
I've seen it fairly commonly, a documentation which repeats quickly what the code does, to give a more literate style. Here we're talking only about one sentence, not overwhelming in my opinion.

It's not about describing the operator though, it's about describing the intent of the code. If it had been more complex (for any reason) than a pipe, a map, a ternary operator and a filtering, I would still keep that phrase, as the statement of the contract this piece of code is trying to achieve.

I personally don't see any problem in "describing a ternary operator with words" (knowing that this is an incorrect shortcut besides). If there is a concrete issue with it, please state it clearly. You just described what I did with your own vision which gives me no useful information.